src/clarity_vm/clarity.rs

```diff
@@ -626,6 +632,56 @@ impl<'a> ClarityBlockConnection<'a> {
         self.cost_track.unwrap()
     }
 
+    pub fn initialize_epoch_2_1(&mut self) -> Result<(), Error> {
+        let mainnet = self.mainnet;
+        let first_block_height = self.burn_state_db.get_burn_start_height();
+        let pox_prepare_length = self.burn_state_db.get_pox_prepare_length();
+        let pox_reward_cycle_length = self.burn_state_db.get_pox_reward_cycle_length();
+        let pox_rejection_fraction = self.burn_state_db.get_pox_rejection_fraction();
+
+        self.as_transaction(|tx_conn| {
```

Comment:
Is there a block cost being applied here?  If so, we probably want to disable it if instantiating `.pox-2` is expensive.

Reply:
Yep -- made it free in d6458eccc939ccf7806470012568b90a5470fda9